sdk/eventhub/Azure.Messaging.EventHubs/src/Core/ObjectExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   The set of extensions for the <see cref="object" />
+    ///   class.
+    /// </summary>
+    ///
+    internal static class ObjectExtensions
+    {
+        /// <summary>
+        ///   Serves as an inverse to the "is" operator, determining whether the <paramref name="instance"/>
+        ///   is NOT of type <typeparamref name="T"/>.
+        /// </summary>
+        ///
+        /// <typeparam name="T">The type of <see cref="Exception"/> test the <paramref name="instance"/> against.</typeparam>
+        ///
+        /// <param name="instance">The instance to consider.</param>
+        ///
+        /// <returns><c>true</c> if the specified instance is NOT of type <typeparamref name="T"/>; otherwise, <c>false</c>.</returns>
+        ///
+        public static bool IsNotType<T>(this object instance) => (!(instance is T));
```

Comment:
For context:  We were starting to build up several negative event filters, sometimes with multiple clauses, that looked like:

```csharp
try 
{
    ...
}
catch (Exception ex) when (!(ex is SomeType))
{
    ...
}
catch (Exception ex) when ((!(ex is OtherType) && (!(ex is ThirdType)))
{
    ...
}
```

With this extension, they transform into a form which I find easier to read and parse:

```csharp
try 
{
    ...
}
catch (Exception ex) when (ex.IsNotType<SomeType>())
{
    ...
}
catch (Exception ex) when ((ex.IsNotType<OtherType>()) && (ex.IsNotType<ThirdType>()))
{
    ...
}
```

Reply:
Yeah, that `typeparam` text is a mistake on my part.   I started off with this scoped to `Exception` types with constraints and then realized that was a silly limitation and generalized.

Reply:
... and it's now back with the change to scope the extension method only to exception scenarios.